### ISSUE TITLE
Support selecting freedom outbound when configuring server

### ISSUE THREAD

**Issue by Vigilans** [v2rayA/v2rayA#621]
### Is the feature related with existing problem?

I want to achieve following use case in v2rayA:

* Create two outbounds `proxy` and `domestic` (If not `proxy` is compulsory, I would use `international` for better symmetry with `domestic`)
* Use `proxy` outbound for non-domestic domains and ips, and `domestic` for CN domains and ips.

Naturally, we have four cases regarding the combination of outbounds:
* `proxy` -> `proxy protocol`, `domestic` -> `freedom`: normal use case.
* `proxy` -> `proxy protocol`, `domestic` -> `proxy protocol`: globally proxied.
* `proxy` -> `freedom`, `domestic` -> `freedom`: proxy disabled.
* `proxy` -> `freedom`, `domestic` -> `proxy tunnel`: access back from non-CN to CN.

The first three use cases are sometimes as frequent as switching between different proxy servers. The only missing pieces for v2rayA to support such scenario is the option to create a freedom outbound when configuring server.

A workaround is to create a socks server that targets a port that will be routed to freedom, or write a hook script to convert a placeholder outbound to freedom outbound.

### Content of the feature

Support selecting freedom outbound when configuring server.

**Comment by mzz2017** [comment]
@Vigilans I'm sorry I do not understand what you mean. I think "direct" outbound is inherit. Do you want to use load balance between normal servers and freedom?

**Comment by Vigilans** [comment]
No, load balancing is irrelevant in this issue. What I am doing is to categorize all domains and IPs into multiple categories, and assign each category with a specific routing strategy, which in this issue, is a target outbound.

For example, we may categorize domains and ips into following categories:
* **domestic**: `geoip:cn` and `geosite:cn`.
* **international**: `geosite:geolocation-!cn`, and non-CN IPs.
* **cernet**: CERNET ips, and domains like `edu.cn`.
* **bilibili**: bilibili related domains.

Then for each category we may expect a different outbound in different scenarios:
* For ordinary use case, **domestic** uses `direct`, **international** uses `proxy`.
* For proxy-disabled use case: **domestic** uses `direct`, **international** uses `direct`.
* For global-proxied use case: **domestic** uses `proxy`, **international** uses `proxy`.
* If we are in edu network, **cernet** uses `direct`, else, **cernet** uses `campus vpn`.
* If we want to watch bilibili hk videos, **bilibili** uses `proxy`, else `direct`.

That is to say, as an outbound strategy, `freedom` is aligned with all other proxy protocols like `vmess` and `ss`. In v2rayA, we may create four outbounds: **proxy** (as **international**), **domestic**, **cernet** and **bilibili**. Then in each outbound, we create serveral servers:

* **proxy**: proxy servers, and freedom
* **domestic**: freedom, and proxy servers
* **cernet**: freedom, and campus vpn outbound (e.g. docker-easyconnect)
* **billability**: freedom, and HK proxy servers.

In this way we could easily manage the specific routing strategy for a specific category in v2rayA (while by manually writing config.json we need at most _2^n_  different configs for _n_ categories). 

Personally I think determining which categories should use `direct` outbound is harder to maintain for v2ray tools.

**Comment by Vigilans** [comment]
For `balancer`, I am not adopting it for load balancing, but for a concept of `Outbound Group`. It is to support an opinion of mine: _routing rules should be static_.

That is, routing strategy should be manually written once, taking all current scenarios into consideration, then kept untouched for rest of time (until new scenario is introduced). Because maintaining and modifying routing rules is cumbersome.

Then how to achieve the switching of category outbounds? If we directly use `proxy` and `direct` outbound, we need to figure out every specific rule that uses `direct` outbound and modify it to `proxy`. Such rules may contain:
* DNS rules. Domestic DNS like `223.5.5.5` normally uses `direct` outbound.
* Domain rules. `geosite:cn`.
* IP rules. `geosite:ip`.

Above three rules may spread across different location in routing rules (e.g. DNS rules at head, domain rules at middle, IP rules at the end), so manually modifying is a tedious work. 

We also cannot easily replace every occurence of `direct` to `proxy`, because some rules must keep direct: private domains and IPs like LAN.

Here `balancer` as `Outbound Group` solves the problem: we define conceptual balancer tags named `domestic`, `international`, `cernet` and `bilibili`. Their actual outbound is not determined in routing rules, but in balancer settings. 

In this way we could keep the routing rules unmodified throughout all scenario, to switch the outbound for a specific category we only need to modify the selected outbound in balancer config: either `direct` or `proxy`. One modification applied to all routing rules.

So, the final routing rules will look like this:
* For private target like `geoip:private` and `geosite:private`, uses `direct` outboundTag.
* For domestic target like `114.114.114.114` dns, `geosite:cn`, `geoip:cn`, uses `domestic` balancerTag.
* For international target like `8.8.8.8` dns, `geosite:geolocation-!cn`, and fallback rule, uses `international`.
* For bilibili target, choose a specific dns (e.g. 223.6.6.6), and `geosite:bilibili`, uses `bilibili`. (by refactoring DNS module, we may use a specific DNS server with a specific tag as inbound for matching)

**Comment by mzz2017** [comment]
@Vigilans I notice your point is like clash group, isn't it? What do you think about RoutingA? Maybe it can satisfy it. Or you want to switch quickly from different RoutingA strategies?

**Comment by Vigilans** [comment]
I am configuring v2rayA on my transparent proxy devices for some of my non-CS friends. I don't mind doing some complex steps to modify the network environment myself, but for friends not familiar with networking and v2ray I expect to provide them with some easy-to-operate ways to switch between different strategies, so I choose v2rayA as the web GUI.

I notice that RoutingA currently needs manual writing, so not the best option for my friends in aforementioned background. If we can prepare multiple RoutingA rules beforehand, and easliy  switch between the rules, then it would be a good choice. 

For my proposed way in this issue, I could implement it using core-hook scripts so this feature request is not a must.

**Comment by mzz2017** [comment]
@Vigilans I see. I'll consider it. Thanks.

**Comment by MarksonHon** [comment]
Closed: no activity for 60 days. Reopen or open a new issue if it still happens on the current release.
